### DISCUSSION TITLE
Prevent File Record Loss: AVAssetWriter movieFragmentInterval

### DISCUSSION THREAD

**Discussion by kenle** [HaishinKit/HaishinKit.swift#1189] [Q&A]
Hi, for IORecorder, may you add the option to set movieFragmentInterval so that the recorded file will save at intervals, in case the app crashes or the user exits the app accidentally, their video file will not be all lost.

An example is shown here:
https://stackoverflow.com/questions/19608424/avassetwriter-file-backup-and-recovery

Relevant file:
https://github.com/shogo4405/HaishinKit.swift/blob/main/Sources/Media/IORecorder.swift

**Comment by shogo4405** [comment]
Sure. I'm busy with work right now, so I'd appreciate it if you could PR.

**Reply by kenle** [reply]
Sure, I will.  Thanks!

**Reply by kenle** [reply]
@shogo4405 It's not as simple as I originally thought.  Adding movieFragmentInterval alone will not do it.  I think you have to save to a temporary file periodically and append to it.  I have to look into it more.

**Reply by xueyongwei** [reply]
Hi，have you found a solution yet? i have tried `movieFragmentInterval` and `shouldOptimizeForNetworkUse`, but did not work.

**Reply by jhays** [reply]
I am also curious about this..
I'm trying to utilize HaishinKit's local recording feature while streaming via RTMP. I'm looking for a solution that would be able to save a local copy of a video that is not interrupted in the event that there are any RTMP disconnections during the stream. In my testing so far, if I cause a network interrupting during a stream, I'm finding that the local recording that HaishinKit is saving is getting broken into 3 .mp4 files. Two of these files are corrupt and cannot be played, while the last file provides a video that starts only after the RTMP stream successfully reconnected.

Is anyone aware of a way to record video locally that just continuously records the camera capture and is completely independent of the RTMP stream's connectivity?

**Reply by shogo4405** [reply]
@jhays I'm creating it to be able to record regardless of network interruptions and verifying that it can record even offline. Could you tell me what specific code you're writing for this?

**Reply by jhays** [reply]
Are you saying that HaishinKit already has the ability to record locally regardless of network interruptions? 
If so, my issue may be something in my implementation then. I will check.

**Reply by shogo4405** [reply]
Yes, it's designed to record regardless of network connectivity. However, there might be cases where issues arise due to differences in testing methods or other factors. Please report any such occurrences to the GitHub Issue as soon as you identify them.

**Comment by shogo4405** [comment]
I tried AVAssetWriter.movieFragmentInterval. I was able to confirm that the file remains even when the task is killed. You can test it with this PR: [#1561](https://github.com/shogo4405/HaishinKit.swift/pull/1561)

> [@shogo4405](https://github.com/shogo4405) It's not as simple as I originally thought. Adding movieFragmentInterval alone will not do it. I think you have to save to a temporary file periodically and append to it. I have to look into it more.

You mentioned that it didn’t simple, but it’s possible that the expected behavior differs between you and me.
